registrations/utils.py

```diff
@@ -1,6 +1,13 @@
 import phonenumbers
+from django.conf import settings
+from wabclient import Client as WABClient
 
 
 def normalise_msisdn(msisdn):
     msisdn = phonenumbers.parse(msisdn, "ZA")
     return phonenumbers.format_number(msisdn, phonenumbers.PhoneNumberFormat.E164)
+
+
+# Short timeout since we're making these requests in the HTTP request
+wabclient = WABClient(url=settings.WHATSAPP_URL, timeout=2)
+wabclient.connection.set_token(settings.WHATSAPP_TOKEN)
```

Comment:
Why not do this in the view?

Reply:
I don't want to do it in the view function, because then we don't get to reuse the same http connection.

It could be done in the view file, but I thought that this is a more general place to do it, in case we need to use the client in other places in the future.

Reply:
But doesn't this mean we'll be creating a new connection with every http request to us anyway?
And if it's here then we're doing it for way more requests than we'll need it for?

Reply:
If it's here, it means that each gunicorn worker should have a connection (after the first request), and that connection gets reused between requests. So that way we don't need to re-establish the connection for every registration, there should already be one (provided it hasn't timed out)

Reply:
Ok I get it. thanks